app/src/main/java/edu/mit/mitmobile2/facilities/FacilitiesFragment.java

```diff
@@ -482,6 +451,88 @@ private void updateProblemViews() {
 
     }
 
+    private Bitmap getBitmap() {
+        Bitmap bitmap = null;
+        try {
+            InputStream is = getActivity().getContentResolver().openInputStream(Uri.parse(photo));
+
+            Display display = getActivity().getWindowManager().getDefaultDisplay();
+            int targetWidth = display.getWidth();
+            int targetHeight = (int) getResources().getDimension(R.dimen.faciliteis_image_view);
+
+            String[] filePathColumn = {MediaStore.Images.Media.DATA};
+            Cursor cursor = getActivity().getContentResolver().query(Uri.parse(photo), filePathColumn, null, null, null);
+            String imagePath;
+            if (cursor == null) {
+                imagePath = Uri.parse(photo).getPath();
+
+            } else {
+                cursor.moveToFirst();
+                int columnIndex = cursor.getColumnIndex(filePathColumn[0]);
+                imagePath = cursor.getString(columnIndex);
+            }
+
+            BitmapFactory.Options options = new BitmapFactory.Options();
+            options.inJustDecodeBounds = true;
+            BitmapFactory.decodeFile(imagePath, options);
+
+            int photoWidth = options.outWidth;
+            int photoHeight = options.outHeight;
+
+            int scaleFactor = 1;
```

Comment:
A cleaner implementation would have something like this (since scaleFactor has to be a power of 2):

while(photoWidth / scaleFactor > targetWidth || photoHeight / scaleFactor > targetHeight) {
scaleFactor *=2;
}